kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -413,24 +414,24 @@ define([
          *    the text?
          * See https://select2.org/dropdown#templating for more details.
          */
-        function formatObjectDisplay(retObj) {
-            if (!retObj.id) {
-                return retObj.text || '';
+        function formatObjectDisplay(obj) {
+            if (!obj.id) {
+                return obj.text || '';
             } else if (dataSource === 'ftp_staging') {
-                return formatFtpStagingDisplay(retObj);
+                return config.isViewOnly ? obj.id : formatFtpStagingDisplay(obj);
```

Comment:
If we're not in the view mode, format the whole expected ftp object, otherwise just return the file name.